crates/ruff_linter/src/rules/flake8_bugbear/rules/batched_without_explicit_strict.rs

```diff
@@ -0,0 +1,101 @@
+use crate::checkers::ast::Checker;
+use crate::fix::edits::add_argument;
+use crate::rules::flake8_bugbear::rules::is_infinite_iterator;
+use crate::settings::types::PythonVersion;
+use ruff_diagnostics::{AlwaysFixableViolation, Applicability, Diagnostic, Fix};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_python_ast::{Arguments, ExprCall};
+
+/// ## What it does
+/// Checks for `itertools.batched` calls without an explicit `strict` parameter.
+///
+/// ## Why is this bad?
+/// By default, if the length of the iterable is not divisible by
+/// the second argument to `itertools.batched`, the last batch
+/// will be shorter than the rest.
+///
+/// Pass `strict=True` to raise a `ValueError` if the batches are of non-uniform length.
+/// Otherwise, pass `strict=False` to make the intention explicit.
+///
+/// ## Example
+/// ```python
+/// itertools.batched(iterable, n)
+/// ```
+///
+/// Use instead:
+/// ```python
+/// itertools.batched(iterable, n, strict=True)
+/// ```
+///
+/// ## Fix safety
+/// This rule's fix is marked as unsafe for `batched` calls that contain
+/// `**kwargs`, as adding a `strict` keyword argument to such a call may lead
+/// to a duplicate keyword argument error.
```

Comment:
I think we should not provide a fix if the call already has a `kwarg` argument or mark it as a manual fix. I'm even leaning towards not raising the diagnostic in that case because we can't know if `strict` is set or not. What do you think? (@AlexWaygood ?)

The fix should be unsafe nonetheless because adding `strict` changes the runtime semantic and the right fix might be to add `strict=False` after all.

Reply:
I strongly agree that we should always mark the fix as unsafe; `strict=True` might not be appropriate for all applications (https://github.com/PyCQA/flake8-bugbear/issues/498#issuecomment-2481323632).

I also agree that we shouldn't offer a fix (or it should be display-only) if there's `kwargs` since there's this risk:

```pycon
>>> kwargs = {"strict": True}
>>> import itertools
>>> itertools.batched([], strict=True, **kwargs)
Traceback (most recent call last):
  File "<python-input-2>", line 1, in <module>
    itertools.batched([], strict=True, **kwargs)
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: itertools.batched() got multiple values for keyword argument 'strict'
```

However, I'd lean towards still emitting the diagnostic if there's kwargs. In most cases, they can probably explicitly pass in `strict=True` or `strict=False`. In the very rare situations where they really need dynamic keyword arguments, I think they can just `# noqa` the rule.

Reply:
Oh, the fix just adds `strict=False`? I'm not sure that's a very useful autofix TBH. Maybe we should just not offer a fix for this rule? The point of the rule is that you should think about your `itertools.batched()` calls and be explicit about whether or not you want the strict behaviour. But you can only tell which is appropriate by doing a manual audit. If Ruff auto-adds `strict=False` everywhere, you haven't really gained anything from the rule

Reply:
(Sorry for not checking what the fix actually did...)

Reply:
I don't really find B905's fix particularly useful either, so I'm not sure it's a great precedent to follow 😆 but interested to hear @MichaReiser's thoughts.

Reply:
I agree that the fix has limited use here and it somewhat defeats the purpose of: Think about what you want. 

Because of that, I went ahead and removed it. It simplifies the implementation